.github/workflows/tests.yml

```diff
@@ -0,0 +1,39 @@
+name: Run tests
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v2
+      with:
+        path: |
+          node_modules
+          vendor/bundle
+        key: ${{ runner.os }}-${{ hashFiles('**/Gemfile.lock') }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-
+    - name: Set up Node 12
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12'
+    - name: Set up Ruby
+      uses: actions/setup-ruby@v1
+      with:
+        ruby-version: 2.5
+    - name: NPM install
+      run: |
+        npm install
+    - name: Bundle install Gems
+      run: |
+        gem install bundler -v 2.1.4
+        bundle config path vendor/bundle
+        bundle install --jobs 4 --retry 3
+    - name: Run UI test suite
+      id: npm_test_run
+      run: |
+        npm run test
+    - name: Run test suite
+      if: ${{ steps.npm_test_run.outcome == 'success' }}
```

Comment:
Does this need to be conditional? They could both run independently, in case JS tests are failing but we are working on ruby stuff...

Reply:
Hm....I would expect at least for `master`, that both JS and Ruby tests pass, so that any work on a feature branch should have the other tests passing. I would be worried about, if you don't make the last step conditional, that only the last system status code is flagged as the returned status of this step, so a failure in the second Ruby tests would get missed.

Reply:
We could also run a completely separate action for Ruby tests (as we are for Ruby gem pushes, and we might for npm pushes some day).

However, it seems like if any of the individual steps fail the whole action is marked as failing, and since you are running them as separate steps, the two steps could run independently, and only if both pass would the action be green?

Reply:
I like separate the separate action idea...